File: python/tmps/tmp2/main.py

def main():
    nums: list = [1, 2, 3, 4]
    result: bool = run(nums)
    
    print(f"Result: {result}")

def run(nums: list[int]) -> bool:
    # hashset
    seen: set[int] = set()

    # iterate given nums
    for val in nums:
        # happy path: if current val in seen, return true
        if val in seen:
            return True
        # update values seen
        seen.add(val)

    return False

if __name__ == "__main__":
    main()

File: typescript/tmps/tmp2/main.js
function main() {
    var nums = [1, 2, 3, 4];
    var target = 7;
    var resp = run(nums, target);
    console.log(resp);
}
function run(nums, target) {
    var seen = new Map();
    for (var i = 0; i < nums.length; i++) {
        var num = nums[i];
        var complement = target - num;
        if (seen.has(complement)) {
            complIdx = seen[complement];
            return [complIdx, i];
        }
        seen.set(num, i);
    }
    return [];
}
main();
